internal/output/ordered_set.go

```diff
@@ -0,0 +1,53 @@
+package output
+
+type OrderedSet[T comparable] struct {
```

Comment:
OrderedSet type does not have `Get(index int)` method. It would be good to add this too. If it is not useful for us, ignore this comment.

Reply:
Let's add it when necessary.